controller/src/main/java/io/pravega/controller/server/security/auth/handler/impl/AclAuthorizerImpl.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.server.security.auth.handler.impl;
+
+import io.pravega.auth.AuthHandler;
+import lombok.NonNull;
+
+class AclAuthorizerImpl extends AclAuthorizer {
+
+    @Override
+    public AuthHandler.Permissions authorize(@NonNull AccessControlList accessControlList, @NonNull String resource) {
+        AuthHandler.Permissions result = AuthHandler.Permissions.NONE;
+
+        String resourceDomain;
+        int indexOfPartsSeparator = resource.indexOf("::");
+        if (indexOfPartsSeparator > 0) {
+            resourceDomain = resource.substring(0, indexOfPartsSeparator);
+        } else {
+            resourceDomain = "prn"; // default
```

Comment:
when we are using the new authorization scheme, then should we even allow for resource domain to be unspecified?

Reply:
Yes, the code above assumes the default `prn` domain. When this code is used as-is in other domains like `prn.schema-registry`, the domain part will automatically be mandatory.

I have two opposite opinions about this. 
* On one hand, allowing for this flexibility could make life a tad easier for the admins, improving usability.
* On the other hand, it's slightly simpler if admins always had to specify it and they didn't have to make an explicit choice (to specify the domain or not). 

In this case, I've favored the former. What do you think?

Reply:
On second thought, we don't actually need to add a default resource domain. This is not the ACE, where domain might be missing. This is really the resource string which is in our control and should always have the domain. I will fix it.

Reply:
I've now pushed the modified code that assumes domain is always present in the resource string used for authorization.